sympy/core/numbers.py

```diff
@@ -1280,7 +1303,11 @@ def __eq__(self, other):
                 return False
             return other.__eq__(self)
         if other.is_Float:
+            # precision is ignored so Float(.1) == Float(.1, 100)
+            # and Float(.1, 100) == Float(.1)
             return bool(mlib.mpf_eq(self._mpf_, other._mpf_))
+        if other.is_Rational:
+            return other.__eq__(self)
```

Comment:
The current implementation of `Rational.__eq__(Float)` will have to be changed if we want to implement the idea that a Float is equal to exactly one Rational, namely the one that represents it.

Reply:
Please check the current -- I think I have it down to 1:1

Reply:
Most rational numbers are not equal to any Float, only those whose denominator is a power of 2. The current implementation converts Rational `self` to mpf:
https://github.com/sympy/sympy/blob/b024c2aa17f6b8d473d893d82a9f7533c132380b/sympy/core/numbers.py#L1777-L1778
The denominator of (the rational number corresponding to) the converted mpf is then a power of 2. That will make it always equal to some Float, which is not desirable.

We should probably first check that `d = self.q` is not a power of 2 (maybe by something like `d & (d - 1) != 0`) and return False in that case. If it is a power of 2, then it possible to convert it to an mpf with almost no computation and that can be used in the comparison.